### PR TITLE
Feature/issues/105

### PR DESCRIPTION
**Pull request by Mizzick**

### CODE REVIEW

File: Extension/lib/filter/update-service.js [hunk 1]
```diff
@@ -251,6 +261,28 @@ exports.ApplicationUpdateService = {
...
+	 * @returns {exports.Promise}
+	 * @private
+	 */
+	_onUpdateFirefoxStorage: function () {
+		Log.info('Call update to version 2.1.1');
+
+		var dfd = new Promise();
+
+		var items = LS.items;
+		for (var i = 0; i < items.length; i++) {
```
Comment (atropnikov) [inline Extension/lib/filter/update-service.js]: You can use for in ... loop for SimpleStorage.storage. It's simpler and you transfer only existing properties.

File: Extension/browser/firefox/lib/main.js [hunk 2]
```diff
@@ -61,7 +63,6 @@ exports.main = function (options, callbacks) {
         var {Log} = loadAdguardModule('utils/log');
         var {FS} = loadAdguardModule('utils/file-storage');
         if (options.loadReason == 'install' || options.loadReason == 'downgrade') {
-            simpleStorage.storage = Object.create(Object.prototype);
             FS.removeAdguardDir();
```
Comment (atropnikov) [inline Extension/browser/firefox/lib/main.js]: Cleanup?

File: Extension/lib/filter/update-service.js (hunk 1 above)
Comment (Mizzick) [inline Extension/lib/filter/update-service.js]: will it work with
delete SimpleStorage.storage[items[i]];

File: Extension/browser/firefox/lib/utils/local-storage.js [hunk 3]
```diff
@@ -22,7 +22,22 @@ var Log = require('utils/log').Log;
  */
 var LS = exports.LS = {
 
-	storage: SimpleStorage.storage,
+	storage: SimplePrefs.prefs,
+
+	items : ['block-list-domains',
```
Comment (atropnikov) [inline Extension/browser/firefox/lib/utils/local-storage.js]: This is redundant property.

File: Extension/lib/filter/update-service.js (hunk 1 above)
Comment (atropnikov) [inline Extension/lib/filter/update-service.js]:
```javascript
var ss = SimpleStorage.storage;
 for (var k in ss) {
  if (ss.hasOwnProperty(k)) {
   var v = ss[k];
   LS.setItem(k, v);
  }
}
SimpleStorage.storage = Object.create(Object.prototype);
```

Comment (atropnikov) [inline Extension/lib/filter/update-service.js]: Or simpler "Properties of the object being enumerated may be deleted during enumeration."

**Pull request merged**